ls1intum/Artemis
# `Assessment`:  Fix an issue during subsequent saving of text exercise assessments

### Checklist
#### General


#### Server

### Motivation and Context
This PR fixes the bug while the assessment of text exercises are subsequently saved, closes #4962.

### Description
There is a foreign key on the TextBlock that references a Feedback entity. When the feedback is removed, due to the constraint in the DB, the foreign key is set to null. If we do not refetch TextBlocks introduced in this PR, the save all method tries to delete non existing records. Also the code quality of DB call is improved (reduced number of calls).

### Steps for Testing

Prerequisites:
- 1 Tutor
- 1 Student
- 1 Text Exercise

1. Log in to Artemis as student
2. Participate in a text exercise
3. As tutor, assess the student's submission by clicking save subsequent times.
4. Verify that the issue introduced in [#4962](https://github.com/ls1intum/Artemis/issues/4962) does not exist.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Not changed.

### Screenshots
![image](https://user-images.githubusercontent.com/67344158/165817389-e805d59c-98a6-43d5-bd57-2f75639bf4d2.png)
